modules/040-node-manager/templates/node-group/_bootstrap.tpl

```diff
@@ -0,0 +1,48 @@
+{{- define "bootstrap_script" -}}
+#!/usr/bin/env bash
+
+set -Eeuo pipefail
+
+function detect_bundle() {
+  {{- .Files.Get "candi/bashible/detect_bundle.sh" | nindent 2 }}
+}
+
+bundle="$(detect_bundle)"
+
+function get_bootstrap() {
+  token="$(</var/lib/bashible/bootstrap-token)"
+  node_group_name="{{ .nodeGroupName }}"
+
+  bootstrap_bundle_name="$bundle.$node_group_name"
+
+  while true; do
+    for server in {{ .apiserverEndpoints | join " " }}; do
+      url="https://$server/apis/bashible.deckhouse.io/v1alpha1/bootstrap/$bootstrap_bundle_name"
+      if curl -s -f -x "" -X GET "$url" --header "Authorization: Bearer $token" --cacert "/var/lib/bashible/ca.crt"
+      then
+        return 0
+      else
+        >&2 echo "failed to get bootstrap $bootstrap_bundle_name with curl https://$server..."
+      fi
+    done
+    sleep 10
+  done
+}
+
+bootstrap_object="$(get_bootstrap)"
+export bootstrap_object
```

Comment:
why we use environment here ?

Reply:
To avoid any problems with string interpolation both in bash and python.